montepy/numbered_mcnp_object.py

```diff
@@ -23,3 +27,39 @@ def old_number(self):
         :rtype: int
         """
         pass
+
+    def clone(self, starting_number=1, step=1):
+        """
+        Create a new independent instance of this object with a new number.
+
+        This relies mostly on ``copy.deepcopy``.
+
+        :param starting_number: The starting number to request for a new object number.
+        :type starting_number: int
+        :param step: the step size to use to find a new valid number.
+        :type step: int
+        :returns: a cloned copy of this object.
+        :rtype: type(self)
+
+        """
+        if not isinstance(starting_number, int):
+            raise TypeError(f"Starting_number must be an int. {starting_number} given.")
+        if not isinstance(step, int):
+            raise TypeError(f"step must be an int. {step} given.")
+        if starting_number <= 0:
+            raise ValueError(f"starting_number must be >= 1. {starting_number} given.")
+        if step <= 0:
+            raise ValueError(f"step must be >= 1. {step} given.")
+        ret = copy.deepcopy(self)
+        if self._problem:
+            ret.link_to_problem(self._problem)
+            collection_type = montepy.MCNP_Problem._NUMBERED_OBJ_MAP[type(self)]
+            collection = getattr(self._problem, collection_type.__name__.lower())
+            ret.number = collection.request_number(starting_number, step)
```

Comment:
If I follow correctly:

`ret.number` necessarily has a setter because of `@make_prop_val_node("_number", ...)`?

Reply:
That's correct, and it is set in the base class `NumberedMCNPObject`